src/python/bcc/__init__.py

```diff
@@ -155,6 +160,63 @@ def generate_auto_includes(cls, program_words):
                         headers += "#include <%s>\n" % header
         return headers
 
+    @classmethod
+    def revise_header_files(self, text):
+        """
+        As for some architectures(especially for arm64), their linux kernel headers files
+        might contain assemble codes and specific register name.
+        However, the clang compiler doesn't recoginize them, and will fail to compile them.
+        Therefore, in order to make the compiler happy, it tries to avoid including them in header files. 
+
+        """
+
+        archtype = os.popen('uname -m').read().strip()
+        if archtype == "aarch64" :
+            extra_header = ""            
+
+            kernel_version=os.popen('uname -r').read().strip()
+            kernel_filelist=[ "/usr/src/kernels/"+kernel_version+"/arch/arm64/include/asm/sysreg.h",
+                              "/usr/src/kernels/"+kernel_version+"/arch/arm64/include/asm/virt.h" ]
+            for filename in kernel_filelist:
+                if os.path.isfile(filename):
+                    filehandle = open(filename)
+       
+                    disable_assemble_flag = True
+                    cur_endif_index = 0
+                    for line in filehandle :
+                        if re.match("#if.*__ASSEMBLY__", line):
```

Comment:
It doesnot remove the "#ifdef __ASSEMBLY". On the contrary, it addes "#if 0" before the "#ifdef __ASSEMBLY". In addition, it also addes the "#endif" after the "#endif" which matches the "#ifdef __ASSEMBLY".